robot-name/robot_name_test.rb

```diff
@@ -29,6 +29,7 @@ def test_reset_name
     robot.reset
     name2 = robot.name
     assert name != name2
+    assert_equal robot.name, name2
```

Comment:
I feel like this tests Ruby itself. You do the assignment on line 30, only to verify that the assignment worked.

I may not understand the edge case that you're trying to avoid. Could you show a code example that has the bug that would pass the old test suite and fail with this one?